Loop/Managers/LoopAppManager.swift

```diff
@@ -0,0 +1,307 @@
+//
+//  LoopAppManager.swift
+//  Loop
+//
+//  Created by Darin Krauss on 2/16/21.
+//  Copyright © 2021 LoopKit Authors. All rights reserved.
+//
+
+import UIKit
+import LoopKit
+import LoopKitUI
+
+public protocol ViewControllerProvider: AnyObject {
```

Comment:
I agree. I don't really like the view controller provider pattern here either. However, there are some problems with doing what you propose above.

When onboarding is active, it is responsible for the root view controller (as provided by the current instance of `OnboardingUI`). Once onboarding is complete, another root view controller (an instance of `RootNavigationController`) is created to contain the `StatusTableViewController` home screen.

There are two managers that need the current root view controller, both dealing with the display of alerts, `AlertManager` (specifically `InAppModalAlertPresenter` owned by `AlertManager`) and `DeviceDataManager` (specifically `DeliveryUncertaintyAlertManager` owned by `DeviceDataManager`).

Both of these manager are needed both *during* onboarding (when onboarding controls the root view controller) and *after* onboarding (when the root view controller `RootNavigationController` contains the `StatusTableViewController`).

Any ideas?

Okay, how about this for an idea... Since this is really only used for presenting alerts, what if we changed the protocol to something like:
```
protocol AlertPresenter {
    func present(_ viewControllerToPresent: UIViewController, animated: Bool, completion: (() -> Void)?)
    func dismiss(animated: Bool, completion: (() -> Void)?)
}
```
The `LoopAppManager` would implement this protocol and use the current root view controller (from the `UIWindow` it knows about).

I think this is a much cleaner protocol. The only (big) problem with this idea is that the `AlertPresenter` name is already in use by an `Alert`-related protocol used in a good number of places (though it appears to issue and retract alerts, not actually present). So, what other name should we use for this new protocol?

Thoughts?

Reply:
FWIW I'm fine if you want to rename that other protocol (but it will touch a bunch of files/repos).  I'd suggest `AlertDisplayer` or maybe just `Alerter`...

Reply:
@ps2 Thoughts? It'd be great to get this hammered out today so I can get this merged before tomorrow.

Reply:
What I was suggesting was keeping the (current dev code) behavior of presenting the onboarding on top of the root loop view controller.

Reply:
Hmm, let me explore that.